src/main/java/com/codeus/winter/config/AbstractAutowireCapableBeanFactory.java

```diff
@@ -0,0 +1,115 @@
+package com.codeus.winter.config;
+
+import com.codeus.winter.annotation.Autowired;
+import com.codeus.winter.exception.BeanFactoryException;
+import com.codeus.winter.util.ClassUtils;
+import jakarta.annotation.Nullable;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * An extension of {@link BeanFactory} that provides extra logic for bean resolving and autowiring.
+ */
+public abstract class AbstractAutowireCapableBeanFactory implements BeanFactory {
+    private final ConstructorResolver constructorResolver;
+
```

Comment:
I don't see any other classes that extend AbstractAutowireCapableBeanFactory, so it shouldn't be an abstract class. Also I don't see other possibilities in the future to use this class for other implementation since it is a core class of framework. 
Looks weird that ConstructorResolver can deal with different implementations of BeanFactory. There is only one main flow for this framework.